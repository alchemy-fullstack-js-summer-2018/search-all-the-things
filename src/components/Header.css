:local(.header) {

  display: flex;
  font-family: 'Roboto', sans-serif;
  padding: 5px;
  align-items: center;

  #nav-ul {
    list-style: none;
    display: flex;
    padding-top: 5px;
  }

  li {
    padding-right: 10px;
  }

  a {
    text-decoration: none;
    color: white;
    font-size: 1.5em;
  }

  a:hover {
    color: yellow;
  }

}