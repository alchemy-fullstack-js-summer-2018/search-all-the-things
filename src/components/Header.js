import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import style from './Header.css';

class Header extends Component {

  render() {
    return (
      <header className={style.header}>
        <div>
          <i className="far fa-moon"></i>
          <Link to="/"><h1>Mystical Tutor</h1></Link>
        </div>
        <nav>
          <ul id="nav-bar">
            <li>
              <Link to="/results?page=1&pageSize=18">Search</Link>
            </li> 
            <li>
              <Link to="/favorites">Favorites</Link>
            </li> 
          </ul>
        </nav>  
      </header>
    );

  }
    
}
export default Header;