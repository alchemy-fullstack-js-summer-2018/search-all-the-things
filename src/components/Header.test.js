import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header';
import toJSON from 'enzyme-to-json';

describe('Header', () => {

  it('renders as designed', () => {
    // shallow render is "visualization of what got render"
    const wrapper = shallow(<Header onSearch={() => {}}/>);
    // make sure it is still doing the same thing
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});