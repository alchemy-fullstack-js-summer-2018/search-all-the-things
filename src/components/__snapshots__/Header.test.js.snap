// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header>
  <div>
    <i
      className="far fa-moon"
    />
    <Link
      replace={false}
      to="/"
    >
      <h1>
        Mystical Tutor
      </h1>
    </Link>
  </div>
  <nav>
    <ul
      id="nav-bar"
    >
      <li>
        <Link
          replace={false}
          to="/results?page=1&pageSize=18"
        >
          Search
        </Link>
      </li>
      <li>
        <Link
          replace={false}
          to="/favorites"
        >
          Favorites
        </Link>
      </li>
    </ul>
  </nav>
</header>
`;
